cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -165,7 +166,17 @@ func getAbapCommunicationArrangementInfo(config abapEnvironmentPullGitRepoOption
 
 	if config.Host != "" {
 		// Host, User and Password are directly provided
-		connectionDetails.URL = "https://" + config.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+		matchedkey, error := regexp.MatchString("[hH][tT][tT][pP][sS]://*", config.Host)
+		if error == nil {
+			if matchedkey {
+				connectionDetails.URL = config.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+			} else {
+				connectionDetails.URL = "https://" + config.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+			}
+		} else {
+			return connectionDetails, errors.New("Error occured while parsing the host parameter. Please check if this parameter has been seet correctly")
+		}
+
```

Comment:
It is a best practice to keep the depth of if statements as low as possible. I would suggest the following, which would also eliminate the else block:

```
if error != nil {
   return
}
if matched Key {
...
```

Reply:
Done.